src/Razor/test/Microsoft.CodeAnalysis.Remote.Razor.Test/Serialization/TagHelperResolutionResultSerializationTest.cs

```diff
@@ -35,99 +26,44 @@ public TagHelperResolutionResultSerializationTest(ITestOutputHelper testOutput)
-        var current = new DirectoryInfo(AppContext.BaseDirectory);
-        while (current != null && !File.Exists(Path.Combine(current.FullName, testFileName)))
-        {
-            current = current.Parent;
-        }
-
-        var tagHelperFilePath = Path.Combine(current.FullName, testFileName);
-        var buffer = File.ReadAllBytes(tagHelperFilePath);
-        var serializer = new JsonSerializer();
-        serializer.Converters.Add(TagHelperDescriptorJsonConverter.Instance);
-        serializer.Converters.Add(TagHelperResolutionResultJsonConverter.Instance);
+        var bytes = TestResources.GetResourceBytes(TestResources.BlazorServerAppTagHelpersJson);
 
-        IReadOnlyList<TagHelperDescriptor> deserializedTagHelpers;
-        using (var stream = new MemoryStream(buffer))
-        using (var reader = new JsonTextReader(new StreamReader(stream)))
+        IReadOnlyList<TagHelperDescriptor> tagHelpers;
+        using (var stream = new MemoryStream(bytes))
+        using (var reader = new StreamReader(stream))
         {
-            deserializedTagHelpers = serializer.Deserialize<IReadOnlyList<TagHelperDescriptor>>(reader);
+            tagHelpers = JsonDataConvert.DeserializeData(reader,
+                static r => r.ReadArrayOrEmpty(
+                    static r => ObjectReaders.ReadTagHelper(r, useCache: false)));
         }
 
-        var expectedResult = new TagHelperResolutionResult(deserializedTagHelpers, Array.Empty<RazorDiagnostic>());
+        var expectedResult = new TagHelperResolutionResult(tagHelpers, Array.Empty<RazorDiagnostic>());
+
+        var serializer = new JsonSerializer { Converters = { TagHelperResolutionResultJsonConverter.Instance } };
 
         // Act
-        MemoryStream serializedStream;
-        using (serializedStream = new MemoryStream())
-        using (var writer = new StreamWriter(serializedStream, Encoding.UTF8, bufferSize: 4096))
+        using var writeStream = new MemoryStream();
+
+        // Serialize the result to a stream
+        using (var writer = new StreamWriter(writeStream, Encoding.UTF8, bufferSize: 4096, leaveOpen: true))
         {
             serializer.Serialize(writer, expectedResult);
         }
 
-        TagHelperResolutionResult deserializedResult;
-        var reserializedStream = new MemoryStream(serializedStream.GetBuffer());
-        using (reserializedStream)
-        using (var reader = new JsonTextReader(new StreamReader(reserializedStream)))
-        {
-            deserializedResult = serializer.Deserialize<TagHelperResolutionResult>(reader);
-        }
+        // Deserialize the result from the stream we just serialized to.
+        writeStream.Seek(0, SeekOrigin.Begin);
 
-        // Assert
-        Assert.Equal(expectedResult, deserializedResult, TagHelperResolutionResultComparer.Default);
-    }
+        TagHelperResolutionResult? actualResult;
 
-    [Fact]
-    public void TagHelperDescriptor_CanReadCamelCasedData()
```

Comment:
📝 I removed this test for the same reason as I described above. It doesn't actually test what it says it's testing, and the thing it's testing is irrelevant anyway. :smile: